mlflow/tracking/client.py

```diff
@@ -1010,6 +1010,62 @@ def log_dict(self, run_id, dictionary, artifact_file):
                 else:
                     json.dump(dictionary, f, indent=2)
 
+    def log_figure(self, run_id, figure, artifact_file):
+        """
+        Log a figure as an artifact. The following figure objects are supported:
+
+        - `matplotlib.figure.Figure`_
+        - `plotly.graph_objects.Figure`_
+
+        .. _matplotlib.figure.Figure:
+            https://matplotlib.org/api/_as_gen/matplotlib.figure.Figure.html
+
+        .. _plotly.graph_objects.Figure:
+            https://plotly.com/python-api-reference/generated/plotly.graph_objects.Figure.html
+
+        :param run_id: String ID of the run.
+        :param figure: Figure to log.
+        :param artifact_file: The run-relative artifact file path in posixpath format to which
+                              the figure is saved (e.g. "dir/file.png").
+
+        .. code-block:: python
+            :caption: Matplotlib Example
+
+            import mlflow
+            import matplotlib.pyplot as plt
+
+            fig, ax = plt.subplots()
+            ax.plot([0, 1], [2, 3])
+
+            run = client.create_run(experiment_id="0")
+            client.log_figure(run.info.run_id, fig, "figure.png")
+
+        .. code-block:: python
+            :caption: Plotly Example
+
+            import mlflow
+            from plotly import graph_objects as go
+
+            fig = go.Figure(go.Scatter(x=[0, 1], y=[2, 3]))
+
+            run = client.create_run(experiment_id="0")
+            client.log_figure(run.info.run_id, fig, "figure.html")
+        """
+        try:
+            import plotly
+            from matplotlib import pyplot as plt
+            from plotly import graph_objects as go
+        except ImportError as ie:
+            print("Failed to import required libraries: {}".format(ie))
```

Comment:
I'll update this line.

Reply:
Is there a list of log_* calls that are going to be added in an Issue somewhere? Can you link me to it just to understand what is coming? It looks like a lot of mlflow-extend is graduating into MLflow :) ?

Reply:
Hi @eedeleon, we're planning to add one more PR which is `log_image`.

Reply:
`log_image` takes an object that represents an image (such as `numpy.ndarray`), serializes it, and log it as an artifact.